src/main/java/de/tum/cit/aet/artemis/programming/service/BuildLogEntryService.java

```diff
@@ -359,25 +407,86 @@ public void deleteOldBuildLogsFiles() {
         if (!profileService.isSchedulingActive()) {
             return;
         }
+
         log.info("Deleting old build log files");
-        ZonedDateTime now = ZonedDateTime.now();
 
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(buildLogsPath)) {
-            for (Path file : stream) {
-                ZonedDateTime lastModified = ZonedDateTime.ofInstant(Files.getLastModifiedTime(file).toInstant(), now.getZone());
-                if (lastModified.isBefore(now.minusDays(expiryDays))) {
-                    Files.deleteIfExists(file);
-                    log.info("Deleted old build log file {}", file);
-                }
-            }
+        try {
+            deleteRecursively(buildLogsPath);
         }
         catch (IOException e) {
             log.error("Error occurred while trying to delete old build log files", e);
         }
     }
 
-    public boolean buildJobHasLogFile(String buildJobId) {
-        Path logPath = buildLogsPath.resolve(buildJobId + ".log");
+    private void deleteRecursively(Path path) throws IOException {
```

Comment:
The method name `deleteRecursively()` seems a bit generic, given that it also checks if the files are expired. Could you be more specific here? e.g. deleteExpiredBuildLogFilesRecursively or something similar?